samples/client/others/csharp-complex-files/src/Org.OpenAPITools/Client/ApiClient.cs

```diff
@@ -536,90 +534,80 @@ private ApiResponse<T> Exec<T>(RestRequest request, RequestOptions options, IRea
                 }
+            };
 
-                InterceptResponse(request, response);
+            return ExecClient(getResponse, setOptions, request, options, configuration);
+        }
 
-                var result = ToApiResponse(response);
-                if (response.ErrorMessage != null)
-                {
-                    result.ErrorText = response.ErrorMessage;
-                }
+        private Task<ApiResponse<T>> ExecAsync<T>(RestRequest request, RequestOptions options, IReadableConfiguration configuration, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            Action<RestClientOptions> setOptions = (clientOptions) =>
+            {
+                //no extra options
+            };
 
-                if (response.Cookies != null && response.Cookies.Count > 0)
+            Func<RestClient, RestResponse<T>> getResponse = (client) =>
+            {
+                Func<Task<RestResponse<T>>> action = async () =>
                 {
-                    if (result.Cookies == null) result.Cookies = new List<Cookie>();
-                    foreach (var restResponseCookie in response.Cookies.Cast<Cookie>())
+                    if (RetryConfiguration.AsyncRetryPolicy != null)
                     {
-                        var cookie = new Cookie(
-                            restResponseCookie.Name,
-                            restResponseCookie.Value,
-                            restResponseCookie.Path,
-                            restResponseCookie.Domain
-                        )
-                        {
-                            Comment = restResponseCookie.Comment,
-                            CommentUri = restResponseCookie.CommentUri,
-                            Discard = restResponseCookie.Discard,
-                            Expired = restResponseCookie.Expired,
-                            Expires = restResponseCookie.Expires,
-                            HttpOnly = restResponseCookie.HttpOnly,
-                            Port = restResponseCookie.Port,
-                            Secure = restResponseCookie.Secure,
-                            Version = restResponseCookie.Version
-                        };
-
-                        result.Cookies.Add(cookie);
+                        var policy = RetryConfiguration.AsyncRetryPolicy;
+                        var policyResult = await policy.ExecuteAndCaptureAsync((ct) => client.ExecuteAsync(request, ct), cancellationToken).ConfigureAwait(false);
+                        return DeserializeRestResponseFromPolicy<T>(client, request, policyResult);
                     }
-                }
-                return result;
-            }
+                    else
+                    {
+                        return await client.ExecuteAsync<T>(request, cancellationToken).ConfigureAwait(false);
+                    }
+                };
+                return action().Result;
```

Comment:
Ive seen some issues with the use of .Result to call the async code from the sync code.
It blocks the UI from rendering in a blazor application and can cause other issues by blocking aync code from yielding.


https://jamie-burns.medium.com/why-you-shouldnt-call-result-when-dealing-with-async-code-in-c-affee2142c86#:~:text=Result'%20is%20a%20blocking%20call,t%20blocked%2C%20but%20it's%20not.